### PR TITLE
[VL] Respect spark.gluten.sql.debug in native side

### PR DESCRIPTION
**Pull request by Yohahaha**
Use spark.gluten.sql.debug to enable print static/session config, print json plan.

### CODE REVIEW

**Comment by Yohahaha** [comment]
We could use velox config lib in whole velox dir.

**Comment by Yohahaha** [comment]
@rui-mo @PHILO-HE please help review this patch, thanks!

**Comment by Yohahaha** [comment]
I think CI failure does not related with this patch. @zhztheplayer

**Review by PHILO-HE — DISMISSED** [review]
Looks good! Thanks!

**Comment by Yohahaha** [comment]
```
2023-11-20T08:02:28.5576407Z java.nio.file.FileSystemException: /tmp/blockmgr-e10fee5b-5d5f-4a2d-8e20-42c7dc891628/25: No space left on device
```

@PHILO-HE seems your CI machine has problems.

**Comment by PHILO-HE** [comment]
> ```
> 2023-11-20T08:02:28.5576407Z java.nio.file.FileSystemException: /tmp/blockmgr-e10fee5b-5d5f-4a2d-8e20-42c7dc891628/25: No space left on device
> ```
> 
> @PHILO-HE seems your CI machine has problems.

It should work now. Let's wait for the re-triggered CI jobs.

**Comment by Yohahaha** [comment]
@PHILO-HE please review again, thank you!

**Comment by PHILO-HE** [comment]
Looks good! Please do a rebase again. Thanks!

**Comment by Yohahaha** [comment]
cc @rui-mo @ulysses-you thank you!

**Review by PHILO-HE — APPROVED** [review]
Thanks!

File: cpp/core/compute/Runtime.h [hunk 1]
```diff
@@ -63,26 +63,9 @@ class Runtime : public std::enable_shared_from_this<Runtime> {
...
 
-  void parsePlan(const uint8_t* data, int32_t size) {
-    parsePlan(data, size, {-1, -1, -1});
-  }
-
   /// Parse and cache the plan.
   /// Return true if parsed successfully.
-  void parsePlan(const uint8_t* data, int32_t size, SparkTaskInfo taskInfo) {
-    taskInfo_ = taskInfo;
-#ifdef GLUTEN_PRINT_DEBUG
```
Comment (rui-mo) [inline cpp/core/compute/Runtime.h:74]: Do we want to control debug log through Gluten config in this PR? Previously, it is controlled by the compilation flag, see https://github.com/oap-project/gluten/blob/main/cpp/CMakeLists.txt#L76-L79. Maybe an option is to control the cpp compilation level through scala config.

**Comment by Yohahaha** [comment]
This patch only move some debug info into scala config.

> Otherwise, we should consider removing `RelWithDebInfo` compilation option, and change all logics controlled by `GLUTEN_PRINT_DEBUG` to the new Scala config.

It's what I'm going to do. CI does not check debug build and we encounter some build error in RelWithDebInfo, and it would be great to debug(not equal to cmake debug) without recompile.

**Comment by rui-mo** [comment]
@Yohahaha Makes sense to me. Could you open an issue on removing `RelWithDebInfo` and `GLUTEN_PRINT_DEBUG`?

**Review by rui-mo — APPROVED** [review]

**Comment by Yohahaha** [comment]
> @Yohahaha Makes sense to me. Could you open an issue on removing `RelWithDebInfo` and `GLUTEN_PRINT_DEBUG`?

https://github.com/oap-project/gluten/issues/3794

**Pull request merged**